scripts/helper.sh

```diff
@@ -38,3 +38,67 @@ function fail_if_error() {
     exit $retVal
   fi
 }
+
+function check_image_artifact_exists() {
+  local FILE=$1
+  if [[ ! -f "$FILE" ]]; then
+    echo "Error: Image artifact $FILE does not exist."
+    set_job_status "error"
+    exit 1
+  fi
+}
+
+# Set up the resolv.conf file for the image root
+# This is used to ensure that the image root can resolve DNS names during customization.
+function setup_resolv() {
+    # Copy the container's /etc/resolv.conf to the image root for customization
+    local IMAGE_ROOT_DIR=$1
+    local IMAGE_ROOT_RESOLV=$IMAGE_ROOT_DIR/etc/resolv.conf
+    local TMP_RESOLV=$IMAGE_ROOT_DIR/tmp/resolv.conf
+    if [ -f "$IMAGE_ROOT_RESOLV" -o -L "$IMAGE_ROOT_RESOLV" ]; then
```

Comment:
Is this the check you want to make? If the `-f` test passes, it means that either of the following things is true:

* `$IMAGE_ROOT_RESOLV` is a regular file
* `$IMAGE_ROOT_RESOLV` is a symbolic link whose target is a regular file.

By adding in the `-L`, then you will also have this check pass in cases where:

* `$IMAGE_ROOT_RESOLV` is a symbolic link whose target is not a regular file
* `$IMAGE_ROOT_RESOLV` is a symbolic link whose target does not exist

I am not super familiar with the intended logic, but that seems unlikely to be what you're wanting.